dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -111,32 +127,89 @@ public User listUser(@ApiParam(value = "Username of user to return") @PathParam(
     @UnitOfWork
     @Path("/{userId}")
     @ApiOperation(value = "Get user with id", response = User.class)
-    public User getUser(@ApiParam(value = "User to return") @PathParam("userId") long userId) {
-        return userDAO.findById(userId);
+    public User getUser(@ApiParam(hidden = true) @Auth Token authToken, @ApiParam(value = "User to return") @PathParam("userId") long userId) {
+        User authUser = userDAO.findById(authToken.getUserId());
+        if (!authUser.getIsAdmin() && authUser.getId() != userId) {
+            throw new WebApplicationException(HttpStatus.SC_FORBIDDEN);
+        }
+
+        User user = userDAO.findById(userId);
+        if (user == null) {
+            throw new WebApplicationException(HttpStatus.SC_BAD_REQUEST);
+        }
+        return user;
     }
 
     @GET
     @Timed
     @UnitOfWork
     @Path("/{userId}/tokens")
-    @ApiOperation(value = "Get user with id", response = Token.class, responseContainer = "List")
-    public List<Token> getUserTokens(@ApiParam(value = "User to return") @PathParam("userId") long userId) {
+    @ApiOperation(value = "Get tokens with user id", response = Token.class, responseContainer = "List")
+    public List<Token> getUserTokens(@ApiParam(hidden = true) @Auth Token authToken,
+            @ApiParam(value = "User to return") @PathParam("userId") long userId) {
+        User user = userDAO.findById(authToken.getUserId());
+        if (!user.getIsAdmin() && user.getId() != userId) {
+            throw new WebApplicationException(HttpStatus.SC_FORBIDDEN);
+        }
+
         return tokenDAO.findByUserId(userId);
     }
 
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/{userId}/tokens/github.com")
+    @ApiOperation(value = "Get Github tokens with user id", response = Token.class, responseContainer = "List")
+    public Token getGithubUserTokens(@ApiParam(hidden = true) @Auth Token authToken,
+            @ApiParam(value = "User to return") @PathParam("userId") long userId) {
+        User user = userDAO.findById(authToken.getUserId());
+        if (!user.getIsAdmin() && user.getId() != userId) {
```

Comment:
This predicate here looks copied and pasted over and over again.
Please split this out into a method in case we need to change this en masse (say we add a new role type).